jetty-ee10/jetty-ee10-maven-plugin/pom.xml

```diff
@@ -81,6 +81,11 @@
       <artifactId>jetty-jndi</artifactId>
       <optional>true</optional>
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-maven</artifactId>
+      <optional>true</optional>
```

Comment:
why optional? seems to be mandatory to have it

Reply:
All the others are optional, so I made it the same. But happy to make it mandatory.

Reply:
same dependency is not optional in ee9 :) 
and btw it is really used by the maven plugin

Reply:
I've made ee9 optional too, so that they all match.

And yes, this dependency is where all the classes are that are common across ee8/9/10 versions of the plugin.

Reply:
> I've made ee9 optional too, so that they all match.
> 
> And yes, this dependency is where all the classes are that are common across ee8/9/10 versions of the plugin.

if it's really used and not depending on what the user will use it's mandatory to have it and not optional.
most of the optional dependencies looks wrong to be optional (but it's not something new I agree :))
why jetty-util is not optional whereas jetty-server is optional?

Reply:
I think these were maybe made optional because of the `distro` way of running the plugin: when we set up the fork for jetty home, we need to copy over any `<dependencies>` that are configured on the `<plugin>` so the webapp works. We put out a warning  if there are any dependencies that are jetty dependencies (because you shouldn't be adding any jetty dependencies to the plugin). The thinking might have been that these dependencies of the plugin itself might register as dependencies (of the project that uses the plugin) and therefore trigger the warning.  See the code here: https://github.com/jetty/jetty.project/blob/jetty-12.0.x/jetty-ee10/jetty-ee10-maven-plugin/src/main/java/org/eclipse/jetty/ee10/maven/plugin/AbstractWebAppMojo.java#L541

If you're sure that the dependencies of the plugin _itself_ won't register  as dependencies of the _usage_ of the plugin, then we can probably take these `optional`s out (although the `optional` setting should only affect people who are trying to use the jetty maven plugin project as a dependency on their own project, which doesn't really make a lot of sense).